### DISCUSSION TITLE
how to export onnx for decoder ?

### DISCUSSION THREAD

**Discussion by Jaionfire** [openai/whisper#1919] [Q&A]
i am export onnx for the encoder, but i am facing issue for export the decoder because if i want export decoder in onnx, i should pass the input as kv_cache but kv_cache is dictionary , but onnx.export not accept the dictionary.  
             
                             torch.onnx.export(self.model.decoder,args=(tokens,audio_features,self.kv_cache),f='tiny_en_decoder.onnx',input_names =["tokens","audio_features","kv_cache])

how to export the onnx for the decoder with kv??.

**Comment by phineas-pta** [comment]
there's already many discussion threads about onnx, u can search to see the code

**Reply by Jaionfire** [reply]
In that discussion didn't mention about the kv_cache as a input to the onnx.export for decoder,.
can please help me on this issue.

**Reply by phineas-pta** [reply]
yes there is, read again, there's many discussion threads not 1

also they already exported onnx file ready to use, why u trying to reinvent the wheel ?
